dataset_loaders/videos/movingMNIST.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+import os
+from scipy.ndimage.interpolation import zoom
+
+from dataset_loaders.parallel_loader import ThreadedDataset
+from dataset_loaders.utils_parallel_loader import natural_keys
+
+
+class MovingMNISTDataset(ThreadedDataset):
+    # Adapted from
+    # https://github.com/dbbert/dfn
+
+    name = 'movingMNIST'
+    # The number of *non void* classes
+    non_void_nclasses = 10
+
+    # A list of the ids of the void labels
+    _void_labels = []
+
+    data_shape = (64, 64, 1)
+
+    # The dataset-wide statistics (either channel-wise or pixel-wise).
+    # `extra/running_stats` contains utilities to compute them.
+    # mean = []
+    # std = []
+
+    # A *dictionary* of the form 'class_id: (R, G, B)'. 'class_id' is
+    # the class id in the original data.
+    _cmap = {}
+
+    # A *dictionary* of form 'class_id: label'. 'class_id' is the class
+    # id in the original data
+    _mask_labels = {}
+
+    def __init__(self, which_set='train', nvids=1000,
+                 image_size=64, num_digits=1, digits_sizes=[28, 28],
+                 digits_speed=0.3, change_dir_prob=[0., 0.],
+                 rng=None, seed=1, *args, **kwargs):
+
+        self.data_shape = (image_size,) * 2 + (self.data_shape[2],)
+
+        self.which_set = 'validation' if which_set == 'valid' else which_set
+        self.nvids = nvids
+        self.image_size_ = image_size
+        self.row_ = 0
+        self.num_digits_ = num_digits
+        self.num_channels_ = 1
+        self.digits_sizes_ = digits_sizes
+        self.num_digits_ = num_digits
+        self.batch_size_ = 1
+        self.set_has_GT = False
+        self.seed = seed
+        self.digits_speed = digits_speed
+        self.change_dir_prob = change_dir_prob
+        self.indices_ = range(nvids)
+        self._rng = rng if rng else np.random.RandomState(seed)
+        # self.path = self.shared_path
+        import h5py
+        try:
+            f = h5py.File(os.path.join(self.path, 'mnist.h5'))
+        except:
+            raise RuntimeError('Failed to load dataset file')
+
+        self.data_ = f[self.which_set].value[:nvids].reshape(-1, 28, 28)
+        f.close()
+
+        super(MovingMNISTDataset, self).__init__(*args, **kwargs)
+
+    def get_names(self):
+        """Return a dict of names, per prefix/subset."""
+        return {'default': range(self.nvids*self.seq_length)}
+
+    def Overlap(self, a, b):
+        """Put b on top of a."""
+        return np.maximum(a, b)
+
+    def get_random_trajectory(self, batch_size):
```

Comment:
Please rename to `_get_random_trajectories`, since it's a private function and returns multiple values.